cmake/app/boilerplate.cmake

```diff
@@ -247,7 +247,18 @@ else() # NOT FIRST_BOILERPLATE_EXECUTION
 
   # Have the child image select the same BOARD that was selected by
   # the parent.
-  set(BOARD ${CACHED_BOARD})
+  # Unless parent was "ns" in which case we assume that the child images are
+  # all are secure, and should be build using the secure version of the board.
+  # It is assumed that only the root app will be built as non-secure.
+  # This is not a valid assumption as there might be multiple non-secure
+  # images defined. With this technique, it is not possible to have multiple
+  # images defined as non-secure.
+  # TODO: Allow multiple non-secure images by using Kconfig to set the
```

Comment:
BTW, isnt' the BSDLIB built as non-secure, too?

Reply:
bsdlib is an app library, isn't it?

Reply:
Yes, as far as I know.

Reply:
Then it will be built as non-secure in the app image.

Reply:
thanx